src/main/java/com/teamknp/hotel/services/KeyStatusService.java

```diff
@@ -1,7 +1,36 @@
 package com.teamknp.hotel.services;
 
+import com.teamknp.hotel.entity.KeyStatus;
+import com.teamknp.hotel.entity.Reservation;
+import com.teamknp.hotel.repository.KeyStatusRepository;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDateTime;
+import java.util.List;
+
 @Service
 public class KeyStatusService {
+    @Autowired
+    KeyStatusRepository keyStatusRepository;
+
+    public void giveOutKey(Reservation reservation, LocalDateTime when) {
+        KeyStatus keyStatus = new KeyStatus();
+        keyStatus.setRoom(reservation.getRoom());
+        keyStatus.setTimeGiven(when);
+        keyStatusRepository.save(keyStatus);
+    }
+
+    public void returnKey(KeyStatus keyStatus, LocalDateTime when) {
+        keyStatus.setTimeReturned(when);
```

Comment:
Te pole `when` jest dziwne. Raczej zawsze będzie mieć jedną wartość, a więc nie trzeba tej funkcji parametryzowąc.

Reply:
Jedną wartość? Czyli 'dzisiaj'?
To troszkę ogranicza opcje CRUD-a. Też się zastanawiałem, czy nie lepiej dać `now()`, ale nie doszedłem do jakiejś sensownej konkluzji, więc zrobiłem to elastycznie.

Reply:
Zrobię to w drugim PR, który już będzie zajmować się (pod)widokiem kluczy i płatności.